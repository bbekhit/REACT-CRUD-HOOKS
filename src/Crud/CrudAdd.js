import React from "react";
import CrudForm from "./CrudForm";
import { connect } from "react-redux";
import { addNote } from "../actions/crudActions";

export const CrudAdd = props => {
  const onSubmit = data => {
    props.addNote(data);
  };
  return (
    <div>
      <CrudForm onSubmit={data => onSubmit(data)} id="add" />
    </div>
  );
};

export default connect(null, { addNote })(CrudAdd);
