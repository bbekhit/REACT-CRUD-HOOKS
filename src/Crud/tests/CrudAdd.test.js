import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import notes from "./mockData";
import { CrudAdd } from "../CrudAdd";

describe("Add Note", () => {
  let wrapper;
  let addNote;
  beforeEach(() => {
    addNote = jest.fn();
    wrapper = shallow(<CrudAdd addNote={addNote} />);
  });

  it("renders correctly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("handle onSubmit correctly", () => {
    wrapper.find("#add").prop("onSubmit")(notes[0]);
    expect(addNote).toHaveBeenLastCalledWith(notes[0]);
  });
});
