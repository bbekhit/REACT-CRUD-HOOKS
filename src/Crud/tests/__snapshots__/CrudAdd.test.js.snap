// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Note renders correctly 1`] = `
<div>
  <Connect(withRouter(CrudForm))
    id="add"
    onSubmit={[Function]}
  />
</div>
`;
